libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -176,6 +177,24 @@ void CodeTransform::operator()(VariableDeclaration const& _varDecl)
 	yulAssert(m_scope, "");
 
 	size_t const numVariables = _varDecl.variables.size();
+
+	// Special case of ``let x := y`` where ``y`` is part of the current scope and this is the last reference to ``y``.
+	// In this case just delete y and assign x y's stack slot.
+	if (Identifier* rhsIdentifier = std::get_if<Identifier>(_varDecl.value.get()))
+		if (Scope::Variable* rhs = std::get_if<Scope::Variable>(m_scope->lookup(rhsIdentifier->name, false)))
+			if (numReferences(*rhs) == 1)
+			{
+				yulAssert(_varDecl.variables.size() == 1, "");
+				YulString lhsName = _varDecl.variables.front().name;
+				auto& lhs = std::get<Scope::Variable>(m_scope->identifiers.at(lhsName));
+				if (!unreferenced(lhs))
```

Comment:
What exactly is this protecting it from?

Reply:
It would probably be fine not to exclude this case - but it's the case where we have ``let x := y`` and ``x`` never being used, so we actually don't need to assign any slot to ``x`` ever - this is handled as special case below - I haven't really looked through it entirely, but I think that special handling is better in that case. It's definitely worth at least a comment, though, that's true.

Reply:
Actually, reading through it a bit more, I think I'll remove this...

The special handling below will ``DUP`` y and immediately ``POP`` again, if I'm reading it correctly - so doing nothing is still preferable to that - and the slot of ``y`` will still be freed - it will then be ``x``s slot, but ``x`` is as unused as ``y`` would have been. I'll try it.

Reply:
Ah ok, so in that case you just reuse the slot anyway and let the rest of the code delete `x` because it's unused

Reply:
Yep - and we should save another DUP POP (even though that'll probably be removed by the old optimizer later anyways). Seems to run through the tests in any case, so I'll remove this - I'll think about adding a test for it as well.

Reply:
Sounds good!